pokemongo_bot/walkers/polyline_generator.py

```diff
@@ -94,7 +94,7 @@ def __init__(self, origin, destination, speed, google_map_api_key=None):
         self._step_keys = sorted(self._step_dict.keys())
         self._last_step = 0
 
-        self._nr_samples = int(min(self.get_total_distance() / self.speed + 1, 512))
+        self._nr_samples = int(min(1 + self.get_total_distance() / max(self.speed, 1), 512))
```

Comment:
Looked a bit in depth... min nr_sample needs to be 2 in order to have elevations returned. otherwise is an invalid response